src/models/serverless.ts

```diff
@@ -55,15 +55,26 @@ export interface ServerlessAzureProvider {
   virtualNetwork?: ResourceConfig;
   armTemplate?: ArmTemplateConfig;
   keyVaultConfig?: AzureKeyVaultConfig;
-  runtime: string;
+  /** Runtime setting within `serverless.yml` */
+  runtime: Runtime;
+  /** Only to be used internally. Derived from `runtime` property */
   functionRuntime?: FunctionRuntime;
+  os?: FunctionAppOS;
 }
 
 export enum FunctionAppOS {
   WINDOWS = "windows",
   LINUX = "linux"
 }
 
+export enum Runtime {
```

Comment:
Seems like we have a few enums now all with similar key/values?  Anyway to simplify this?

Reply:
Made some pretty significant updates to simplify this. No longer have generated `functionRuntime` property